### ISSUE TITLE
Make generated project fully compatible with Edge

### ISSUE THREAD

**Issue by JimMorrison723** [webextension-toolbox/generator-web-extension#20]
## Expected Behavior

Full Edge support out-of-the-box.

## Current Behavior

After generating extension, we have to manually add `author ` field to manifest.json and `persistent` property to the background page object.

Also, Edge doesn't support promises ([yet](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/9421085/)), maybe use this [polyfill version](https://github.com/mozilla/webextension-polyfill/pull/114)?

## Steps to Reproduce (for bugs)

1. generate extension
2. run it
3. try to load it into Edge

## Suggestion

1., Add persistent to manifest file by default (if we select to add background page), like this:
```
"background": {
    "scripts": [
        "scripts/background.js"
    ],
    "__chrome|edge|opera__persistent": false
},
```
[Documentation](https://developer.chrome.com/extensions/background_pages#manifest)
[Browser compatibility](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/background) (not supported in firefox)

2., Simply add `author` property to default manifest.json

[Documentation](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/author) (**This key is mandatory in Microsoft Edge.**)

3., Polyfill to Edge version
If i simply add the polyfill, `browser` will be undefined. I assume using [this](https://github.com/mozilla/webextension-polyfill/pull/114) fixes the issue. Needs testing.

[Documentation](https://docs.microsoft.com/en-us/microsoft-edge/extensions/api-support/supported-apis) (Microsoft Edge extension APIs use callbacks, not promises.)

## Your Environment

| Tech                      | Version |
|---------------------------|---------|
| generator-webextension    | 0.0.5  |
| node                      | v10.4.0 |
| OS                        | W10|
| browser                   |         |
| etc                       |         |

**Comment by balcsida** [comment]
So, I wrote a task list (because I like task lists):
- [ ] Add persistent to manifest file by default (if we select to add background page)
- [ ] Simply add author property to default manifest.json
- [ ] Polyfill to Edge version

The first two item looks like an easy win to me. But the last one...
It definitely needs some additional investigation and test.

**Comment by balcsida** [comment]
For the last task, I will follow mozilla/webextension-polyfill#154 as well

**Comment by balcsida** [comment]
Depends on mozilla/webextension-polyfill#114

**Comment by balcsida** [comment]
Hi @JimMorrison723,

Because Edge will switch Chromium base, I closed this Issue and PR #26.

For further explanation, please see rpl's comment here:
https://github.com/mozilla/webextension-polyfill/pull/114#issuecomment-451923835

Kind regards,
@balcsida

**Comment by JimMorrison723** [comment]
Hi @balcsida!
Yeah, when i heard about the switch, i was happy that we don't need this after all. Good thing we waited... =)
